libs/community/langchain_community/embeddings/openvino.py

```diff
@@ -0,0 +1,298 @@
+                "Could not import optimum-intel python package. "
+                "Please install it with: "
+                "pip install 'optimum[openvino,nncf]' "
+            ) from e
+        try:
+            # use local model
+            self.ov_model = OVModelForFeatureExtraction.from_pretrained(
+                self.model_name_or_path, **self.model_kwargs
+            )
+        except Exception:
+            # use remote model
+            self.ov_model = OVModelForFeatureExtraction.from_pretrained(
+                self.model_name_or_path, export=True, **self.model_kwargs
+            )
+
+        try:
+            from transformers import AutoTokenizer
+        except ImportError as e:
+            raise ImportError(
+                "Unable to import transformers, please install with "
+                "`pip install -U transformers`."
+            ) from e
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path)
+
+    def _text_length(self, text: Any) -> int:
+        """
+        Help function to get the length for the input text. Text can be either
+        a list of ints (which means a single text as input), or a tuple of list of ints
+        (representing several text inputs to the model).
+        """
+
+        if isinstance(text, dict):  # {key: value} case
+            return len(next(iter(text.values())))
+        elif not hasattr(text, "__len__"):  # Object has no len() method
+            return 1
+        # Empty string or list of ints
+        elif len(text) == 0 or isinstance(text[0], int):
+            return len(text)
+        else:
+            # Sum of length of individual strings
+            return sum([len(t) for t in text])
+
+    def encode(
+        self,
+        sentences: Any,
+        batch_size: int = 4,
+        show_progress_bar: bool = False,
+        convert_to_numpy: bool = True,
+        convert_to_tensor: bool = False,
+        mean_pooling: bool = False,
+        normalize_embeddings: bool = True,
+    ) -> Any:
+        """
+        Computes sentence embeddings.
+
+        :param sentences: the sentences to embed.
+        :param batch_size: the batch size used for the computation.
+        :param show_progress_bar: Whether to output a progress bar.
+        :param convert_to_numpy: Whether the output should be a list of numpy vectors.
+        :param convert_to_tensor: Whether the output should be one large tensor.
+        :param mean_pooling: Whether to pool returned vectors.
+        :param normalize_embeddings: Whether to normalize returned vectors.
+
+        :return: By default, a 2d numpy array with shape [num_inputs, output_dimension].
+        """
+        try:
+            import numpy as np
+        except ImportError as e:
+            raise ImportError(
+                "Unable to import numpy, please install with " "`pip install -U numpy`."
+            ) from e
+        try:
+            from tqdm.autonotebook import trange
```

Comment:
does it really need to be exported from notebook related namespace?

Reply:
done